src/aggregate/reducers/random_sample.c

```diff
@@ -49,19 +49,12 @@ static int sampleAdd(Reducer *rbase, void *ctx, const RLookupRow *srcrow) {
 
 static RSValue *sampleFinalize(Reducer *rbase, void *ctx) {
   rsmplCtx *sc = ctx;
-  RSMPLReducer *r = (RSMPLReducer *)rbase;
-  size_t len = MIN(r->len, sc->seen);
-  RSValue *ret = sc->samplesArray;
-  sc->samplesArray = NULL;
-  return ret;
+  return RSValue_IncrRef(sc->samplesArray);
```

Comment:
Utilizing the refcount mechanism instead of giving away the pointer. It just makes the code flow simpler.
The `Finalize` function gives a reference to the caller, and freeing the reducer instance always includes releasing the instance reference. This is the common way of passing ownership over RSValue in other places in the code as well

Reply:
Is the a comment documenting the ownership transfer?